File: server/src/main/java/com/bomberman/entities/mapobject/Brick.java
package com.bomberman.entities.mapobject;

import com.bomberman.constants.Configuration;
import com.bomberman.entities.Position;

public class Brick extends MapObject {

  public Brick(final Position position) {
    super(position);
  }

  @Override
  public int getTextureWidth() {
    return Configuration.TEXTURE_SIZE;
  }

  @Override
  public int getTextureHeight() {
    return Configuration.TEXTURE_SIZE;
  }
}

File: client/src/entities/mapobject/Brick.ts
import {MapObject} from './MapObject';
import {Adapter} from "../Adapter";
import {Position, PositionAdapter} from "../Position";

export class Brick extends MapObject {

    constructor(position: Position) {
        super(position);
    }
}

export class BrickAdapter implements Adapter<Brick> {
    adapt(item): Brick {
        if (!item) {
            return null;
        }
        const positionAdapter = new PositionAdapter();
        let position = positionAdapter.adapt(item.position)
        return new Brick(position);
    }
}